crates/networking/rpc/engine/blobs.rs

```diff
@@ -110,55 +115,205 @@ impl RpcHandler for BlobsV2Request {
-            for (commitment, (blob, proofs)) in commitments_in_bundle.iter().zip(
-                blobs_in_bundle
-                    .iter()
-                    .zip(proofs_in_bundle.chunks(CELLS_PER_EXT_BLOB)),
-            ) {
-                let current_versioned_hash = kzg_commitment_to_versioned_hash(commitment);
-                if let Some(index) = self
-                    .blob_versioned_hashes
-                    .iter()
-                    .position(|&hash| hash == current_versioned_hash)
-                {
-                    // If the versioned hash is one of the requested we save its corresponding blob and proof in the returned vector. We store them in the same position as the versioned hash was received.
-                    res[index] = Some(BlobAndProofV2 {
-                        blob: *blob,
-                        proofs: proofs.to_vec(),
-                    });
-                }
+    if let Some(current_block_header) = context
+        .storage
+        .get_block_header(context.storage.get_latest_block_number().await?)?
+        && !context
+            .storage
+            .get_chain_config()
+            .is_osaka_activated(current_block_header.timestamp)
+    {
+        // validation requested in https://github.com/ethereum/execution-apis/blob/a1d95fb555cd91efb3e0d6555e4ab556d9f5dd06/src/engine/osaka.md?plain=1#L130
+        return Err(RpcErr::UnsuportedFork(format!(
+            "getBlobsV{} engine only supported for Osaka",
+            version
+        )));
+    };
+
+    let mut res: Vec<Option<BlobAndProofV2>> = vec![None; blob_versioned_hashes.len()];
+
+    for blobs_bundle in context.blockchain.mempool.get_blobs_bundle_pool()? {
+        // Go over all blobs bundles from the blobs bundle pool.
+        let blobs_in_bundle = blobs_bundle.blobs;
+        let commitments_in_bundle = blobs_bundle.commitments;
+        let proofs_in_bundle = blobs_bundle.proofs;
+
+        // Go over all the commitments in each blobs bundle to calculate the blobs versioned hash.
+        for (commitment, (blob, proofs)) in commitments_in_bundle.iter().zip(
+            blobs_in_bundle
+                .iter()
+                .zip(proofs_in_bundle.chunks(CELLS_PER_EXT_BLOB)),
+        ) {
+            let current_versioned_hash = kzg_commitment_to_versioned_hash(commitment);
+            if let Some(index) = blob_versioned_hashes
+                .iter()
+                .position(|&hash| hash == current_versioned_hash)
+            {
+                // If the versioned hash is one of the requested we save its corresponding blob and proof in the returned vector. We store them in the same position as the versioned hash was received.
+                res[index] = Some(BlobAndProofV2 {
+                    blob: *blob,
+                    proofs: proofs.to_vec(),
+                });
             }
         }
-        if res.iter().any(|blob| blob.is_none()) {
-            return Ok(Value::Null);
+    }
```

Comment:
This looks like a lot of work to do for a single RPC call. We should probably, at the very least, hash once when we accept the blob tx to the mempool and keep that hash, rather than hash all of them each time someone calls this endpoint. Looks like a DoS vector to me.

NOTE: this is not necessarily the place to fix it, being already there, it just got it to my attention.
@MegaRedHand should we open an issue for this?